geth/jail/cell_test.go

```diff
@@ -91,43 +99,82 @@ func (s *CellTestSuite) TestCellFetchRace() {
 
 	err := cell.Set("__captureSuccess", func(res otto.Value) { dataCh <- res })
 	s.NoError(err)
+
 	err = cell.Set("__captureError", func(res otto.Value) { errCh <- res })
 	s.NoError(err)
 
-	// run JS code for fetching valid URL
-	_, err = cell.Run(`fetch('` + server.URL + `').then(function(r) {
-		return r.text()
+	fetchCode := `fetch('%s?i=%d').then(function(r) {
+ 		return r.text()
 	}).then(function(data) {
 		__captureSuccess(data)
 	}).catch(function (e) {
 		__captureError(e)
-	})`)
+	})`
+
+	for i := 0; i < requestCount; i++ {
+		_, err = cell.Run(fmt.Sprintf(fetchCode, server.URL, i))
+		s.NoError(err)
+	}
+
+	var mutex sync.Mutex
```

Comment:
didn't check the whole PR yet, but this mutex isn't really needed

Reply:
@dshulyak wouldn't that cause race condition as it makes change on the `expected` variable when a value passed through `dataCh` ? (I got race cond errors when I tried removing it)

Reply:
reading from a channel is a synchronous operation, usually if there is concurrent access to a map error would be smth like this `fatal error: concurrent map writes`. And this is what I am seeing when mutex is removed: 

```
WARNING: DATA RACE
Write at 0x00c420020a20 by goroutine 21:
  runtime.closechan()
      /snap/go/1333/src/runtime/chan.go:313 +0x0
  github.com/status-im/status-go/geth/jail/internal/loop.(*Loop).close()
      /home/ds/status/src/github.com/status-im/status-go/geth/jail/internal/loop/loop.go:74 +0xdf
  github.com/status-im/status-go/geth/jail/internal/loop.(*Loop).Run()
      /home/ds/status/src/github.com/status-im/status-go/geth/jail/internal/loop/loop.go:181 +0x258
  github.com/status-im/status-go/geth/jail.NewCell.func1()
      /home/ds/status/src/github.com/status-im/status-go/geth/jail/cell.go:52 +0x50

Previous read at 0x00c420020a20 by goroutine 64:
  runtime.chansend()
      /snap/go/1333/src/runtime/chan.go:128 +0x0
  github.com/status-im/status-go/geth/jail/internal/loop.(*Loop).Ready()
      /home/ds/status/src/github.com/status-im/status-go/geth/jail/internal/loop/loop.go:135 +0xf6
  github.com/status-im/status-go/geth/jail/internal/timers.newImmediateTimerHandler.func1.1()
      /home/ds/status/src/github.com/status-im/status-go/geth/jail/internal/timers/timers.go:93 +0x56```

Reply:
@dshulyak you're right, I'll debug it tomorrow (in my timezone).

Reply:
@dshulyak It was caused by `TearDownTest` closing the loop channel when there is a task about to send a value to the closed ready channel. I added a 100ms delay at the end of the test to avoid that: https://github.com/status-im/status-go/pull/627/files#diff-4e0a55d0b6f208d64dcebe6c0cb5eabeR136